### ISSUE TITLE
NVDA switches out of application mode when focus lands on links or buttons inside a gridcell element

### ISSUE THREAD

**Issue by Wildebrew** [nvaccess/nvda#18265]
When entering a data grid, NVDA automatically switches to application mode. 
This is good. A data grid in a web application is expected to manage keyboard navigation.

But, an it's a considerable but and I cannot lie, when focus lands on an element with either a link or a button role inside a grid cell in the data grid, NVDA automatically switches back to browse mode. 
That makes grid navigation challenging and treacherous.

### Steps to reproduce:

Using [the APG data grid examples](https://www.w3.org/WAI/ARIA/apg/patterns/grid/examples/data-grids/)

1. Load the page and start navigating with the tab key until you hit a grid example. NVDA obligingly switches to application mode (or whatever you call it these days)
2. Use the arrow keys to navigate the grid until you hit an element with  a link or a button role. The first link you should hit is the "Cash deposit" link in the description column of the first grid.
3. Notice how NVDA instantly switches to browse mode, even if focus is inside an element with role="gridcell" in a data grid.

 ### Actual behavior:

NVDA switches to browse mode when focus lands on a link or button inside a grid cell in a data grid.

### Expected behavior:

NVDA should stay in application mode until focus has left the grid (either by navigating out of it or by activating a control in a grid cell that navigates away from the grid).

### NVDA logs, crash dumps and other attachments:

No crashes or errors involved, this is just confusing out-of-the-box behavior.

#### NVDA installed/portable/running from source:

Latest version of NVDA running on Windows (tested with latest version of Chrome).
This also applies to multiple NVDA versions goin back at least 2 years.

#### NVDA version:

Tested on multiple versions starting wit v2024.1 (not tested on older versions).

#### Windows version:
Tested on multiple Windows 10 and 11 versions.

#### Name and version of other software in use when reproducing the issue:

Chrome browser, multiple recent versions.
(not tested on Firefox but suspect same behavior)

#### Does the issue still occur after restarting your computer?
Certainly.

#### Have you tried any other versions of NVDA? If so, please report their behaviors.
Yes, multiple, all with same result.

#### If NVDA add-ons are disabled, is your problem still occurring?
N/A

#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
Yes
